### ISSUE TITLE
gmusic: Automatically Upload to Google Music on Import

### ISSUE THREAD

**Issue by shuaiscott** [beetbox/beets#2997]
### Feature Request

Uploading all my current music via Beets would result in a lot of duplicate entries. It'd be nice to have the functionality to automatically upload to Google Music when new items are added to the library

**Comment by sampsyo** [comment]
Sounds good! Any hackers in the `gmusic` plugin are hereby invited to pitch in. 😇

**Comment by shuaiscott** [comment]
Took a look at how this would've been implemented and decided to knock it out real quick 😄

**Comment by jsetton** [comment]
@shuaiscott thanks for implementing this change. However, I noticed that the files uploaded during the import are missing some of the metadata updates made by other plugins such as album art images or featured artists added to title. Do you think it would be possible for the upload to happen after all the plugins interacting with the metadata are done?

**Comment by shuaiscott** [comment]
Yeah, I think I noticed this too on my own auto-uploads. However, looking through the docs, it appears that plugins run all at the same time in parallel... meaning that while trying to get artwork and lyrics and etc, it will be uploading to Google Music at the same time. Might be best to just manually upload to Google after all, unless another post-processing plugin feature is added to beets.

**Comment by shuaiscott** [comment]
@jsetton I just tried importing a new song with auto upload turned off. It too was missing some metadata and artwork. Maybe there's something going on with the Python library or with Google's upload service

**Comment by jsetton** [comment]
@shuaiscott Related to the auto upload feature, based on my understanding, you can register the process to run after a specific [event](https://beets.readthedocs.io/en/v1.4.6/dev/plugins.html#listen-for-events). The downside is that it would be a blocking call compared to just having that process run asynchronously during the import stages.

Anyway, I have since moved away from gmusicapi and started to use a new [google-music api wrapper](https://github.com/thebigmunch/google-music) via a shell script.

**Comment by shuaiscott** [comment]
Wow, I didn't realize that was built into plugins now. I'll reopen this and hopefully get this added to auto import after the whole import flow is run. 

```self.register_listener('import', self.autoimport)```

**Comment by wisp3rwind** [comment]
Closing, since there's no more Google Play Music: https://github.com/beetbox/beets/issues/4089
